eth2/simulation/src/simulation.rs

```diff
@@ -15,37 +17,205 @@ where
 }
 
 impl<T: EthSpec> Simulation<T> {
+    pub fn new() -> Self {
+        Self {
+            store: Store::new(),
+        }
+    }
+
     /// Add a new execution environment, return EE index
-    fn create_execution_environment(a: args::CreateExecutionEnvironment) -> EEIndex {
-        unimplemented!();
+    pub fn create_execution_environment(
+        &mut self,
+        a: args::CreateExecutionEnvironment,
+    ) -> Result<EEIndex> {
+        // Create EE struct from args
+        let vl_wasm_code =
+            VariableList::new(a.wasm_code).map_err(|_| Error::MaxLengthExceeded {
```

Comment:
You should be able to use snafu's [`ResultExt::context`](https://docs.rs/snafu/0.6.2/snafu/trait.ResultExt.html#tymethod.context) method for these types of conversions.

Reply:
Same comment as above -- interested to hear your thoughts on my breakdown of the two options.  As mentioned, I have a slight preference for this because it is simpler, uses std library methods, and IMO is more readable.